### ISSUE TITLE
Occasionally blocked on disconnected state

### ISSUE THREAD

**Issue by sunng87** [sunng87/rustmann#9]
```
cargo run --example interactive
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/examples/interactive`
please
Err(IoError(Kind(TimedOut)))
yes
Err(IoError(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }))
miss

```

The library is blocked in connecting state when the first attempt was failed.

**Comment by sunng87** [comment]
```
nsun@thinkless: ~/tmp/proc-macro-error master
$ cargo release patch --skip-publish                                                           [22:01:05]
Release version proc-macro-error-attr 0.3.1? [y/N]
y
Update proc-macro-error-attr to version 0.3.1
Adding to dependencies
[master de56bb5] (cargo-release) version 0.3.1
 2 files changed, 2 insertions(+), 2 deletions(-)
Creating git tag proc-macro-error-attr-v0.3.1
Starting proc-macro-error-attr's next development iteration 0.3.2-alpha.0
Fatal: Invalid TOML file format: error during execution of `cargo metadata`: error: no matching package named `proc-macro-error-attr` found
location searched: /home/nsun/tmp/proc-macro-error/proc-macro-error-attr
prerelease package needs to be specified explicitly
proc-macro-error-attr = { version = "0.3.2-alpha.0" }
required by package `proc-macro-error v0.3.1 (/home/nsun/tmp/proc-macro-error/proc-macro-error)`

FAIL
Err(IoError(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }))
s
[src/client.rs:61] "disconnected" = "disconnected"
```

By adding debug logs, it seems the future stopped on `disconnected` state.

**Comment by sunng87** [comment]
Fixed in ed952a0
